src/visitable.cpp

```diff
@@ -103,9 +119,7 @@ static int has_quality_from_vpart( const vehicle& veh, int part, const quality_i
 
             // does the part provide this quality?
             if( iter != tq.end() && iter->second >= level ) {
-                if( ++qty >= limit ) {
-                    break;
-                }
+                qty = sum_no_wrap( qty, 1 );
```

Comment:
The logic prevents wrapping everywhere, selective enforcement makes the code fragile.